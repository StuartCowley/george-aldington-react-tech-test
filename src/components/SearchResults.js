import React from "react";
import PropTypes from "prop-types";
import "../styles/search-results.css";

function SearchResults({ results }) {
  if (!results.length) {
    return (
      <p className="no-images-message">Sorry, we couldn't find anything.</p>
    );
  }
  return (
    <div className="search-results">
      {results.map((image, index) => {
        return (
          <img
            className="search-results__image"
            src={image}
            alt="placeholder"
            key={index}
          />
        );
      })}
    </div>
  );
}

SearchResults.propTypes = {
  results: PropTypes.array.isRequired,
};
export default SearchResults;
