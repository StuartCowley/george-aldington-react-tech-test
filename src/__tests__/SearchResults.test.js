import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  let resultsMock = ["test1", "test2"];

  it("renders results correctly", () => {
    const { asFragment } = render(<SearchResults results={resultsMock} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly if there are no results", () => {
    resultsMock = [];
    render(<SearchResults results={resultsMock} />);
    const displayMessage = screen.getByText(
      "Sorry, we couldn't find anything."
    );

    expect(displayMessage).toBeInTheDocument();
  });
});
