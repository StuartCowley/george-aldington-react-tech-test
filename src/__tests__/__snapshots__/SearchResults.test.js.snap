// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders results correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="placeholder"
      class="search-results__image"
      src="test1"
    />
    <img
      alt="placeholder"
      class="search-results__image"
      src="test2"
    />
  </div>
</DocumentFragment>
`;
